diff --git a/dwm.c b/dwm.c
index a811928..f215d97 100644
--- a/dwm.c
+++ b/dwm.c
@@ -946,13 +946,26 @@ drawbar(Monitor *m)
 	char ctmp;
 	Client *c;
 
+    /* correction for colours */
+    int correct = 0; 
+    char *xcape = malloc (sizeof (char) * 128);
+    memset(xcape,0,sizeof (char) * 128);
+    for ( ; *ts != '\0' ; ts++) {    
+        if (*ts <= LENGTH(colors)) {
+            sprintf(xcape,"%c",*ts);
+            correct += TEXTW(xcape) - lrpad;
+        }
+    }
+    free(xcape);
+    ts = stext;
+
 	if (!m->showbar)
 		return;
 
 	/* draw status first so it can be overdrawn by tags later */
 	if (m == selmon) { /* status is only drawn on selected monitor */
 		drw_setscheme(drw, scheme[SchemeNorm]);
-		tw = TEXTW(stext) - lrpad + 2; /* 2px right padding */
+		tw = TEXTW(stext) - lrpad + 2 - correct; /* 2px right padding */
 		while (1) {
 			if ((unsigned int)*ts > LENGTH(colors)) { ts++; continue ; }
 			ctmp = *ts;
